dynawo/sources/Modeler/Common/DYNModelMulti.cpp

```diff
@@ -298,13 +296,33 @@ ModelMulti::init(const double& t0) {
 
 void
 ModelMulti::printModel() {
+#ifdef _DEBUG_
   Timer timer("ModelMulti::printModel");
+#endif
   for (unsigned int i = 0; i < subModels_.size(); ++i)
     subModels_[i]->printModel();
 
   connectorContainer_->printConnectors();
 }
 
+void
+ModelMulti::printParameterValues() const {
+#ifdef _DEBUG_
+  Timer timer("ModelMulti::printParameterValues");
+#endif
+  Trace::debug(Trace::parameters()) << "This file is organized as follows: "<< Trace::endline;
+  Trace::debug(Trace::parameters()) << "  -- For each model, initial and dynamic models parameters values before local initialization"<< Trace::endline;
+  Trace::debug(Trace::parameters()) << "  -- For each model, dynamic models parameters values after local initialization"<< Trace::endline;
+  Trace::debug(Trace::parameters()) << "caption: "<< Trace::endline;
+  Trace::debug(Trace::parameters()) << "  -- \"modelica file\"  -> default value specified in modelica model"<< Trace::endline;
+  Trace::debug(Trace::parameters()) << "  -- \"parameters\"     -> value read from parameter file"<< Trace::endline;
+  Trace::debug(Trace::parameters()) << "  -- \"IIDM\"           -> value read from iidm file"<< Trace::endline;
+  Trace::debug(Trace::parameters()) << "  -- \"loaded dump\"    -> value read from initial state file"<< Trace::endline;
+  Trace::debug(Trace::parameters()) << "  -- \"initialisation\" -> value computed by local initialization"<< Trace::endline;
```

Comment:
I would be annoying but isn't there a way to only open the flux once (and test the fact to open the log) and then write the whole log at once?
I would find it more beautiful ;)

Reply:
done :D